File: test-automation-core/src/main/java/com/nortal/test/core/services/hooks/AfterScenarioHook.java
package com.nortal.test.core.services.hooks;

import com.nortal.test.core.services.ScenarioContainer;

/**
 * Interface for performing after scenario cleanup. Any bean that implements this interface will have its cleanup method called during cucumber @After
 * tag.
 */
public interface AfterScenarioHook {
	/**
	 * Perform after scenario operations.
	 */
	void after(ScenarioContainer scenario);
}

File: test-automation-core/src/main/kotlin/com/nortal/test/core/services/hooks/AfterScenarioHook.kt
package com.nortal.test.core.services.hooks

import com.nortal.test.core.services.ScenarioExecutionContext

/**
 * Interface for performing after scenario cleanup. Any bean that implements this interface will have its cleanup method called during cucumber @After
 * tag.
 */
interface AfterScenarioHook {
    /**
     * Perform after scenario operations.
     */
    fun after(scenario: ScenarioExecutionContext?)
}